src/components/molecules/Monaco/useCodeIntel.ts

```diff
@@ -34,31 +34,32 @@ function useCodeIntel(
 
   const applyCodeIntel = () => {
     if (editor && selectedResource) {
-      const {newDecorations, newDisposables} = codeIntel.applyForResource(
-        selectedResource,
-        selectResource,
-        selectFilePath,
-        createResource,
-        resourceMap,
-        fileMap
-      );
-      idsOfDecorationsRef.current = setDecorations(editor, newDecorations);
-      disposablesRef.current = newDisposables;
+      codeIntel
+        .applyForResource(
+          selectedResource,
+          selectResource,
+          selectFilePath,
+          createResource,
+          filterResources,
+          resourceMap,
+          fileMap,
+          editor?.getModel()
+        )
+        .then(({newDecorations, newDisposables}) => {
+          idsOfDecorationsRef.current = setDecorations(editor, newDecorations);
+          disposablesRef.current = newDisposables;
+        });
     }
   };
 
-  useDebounce(
-    () => {
-      clearCodeIntel();
-      applyCodeIntel();
+  useEffect(() => {
```

Comment:
@devcatalin I had to go back to useEffect since useDebounce wasn't calling the cleanup function - perhaps there is some way to combine these two!? to still achieve a debounce on property changes? (since many change quite often)..

Reply:
we can keep the `useEffect` hook and use the `debounce` function from lodash instead

Reply:
thanks I'll try that

Reply:
ok - debounce added - hope I got it right... thanks @devcatalin